[tune] Expose progress reporter to users (#6915)
ray-project/ray: doc/source/tune-usage.rst

Note that some behavior such as writing to files by depending on the current working directory in a Trainable and setting global process variables may not work as expected. Local mode with multiple configuration evaluations will interleave computation, so it is most naturally used when running a single configuration evaluation.

CLI Progress Reporting
----------------------

> By default, Tune reports experiment progress to the command-line at the end of each training iteration as follows.

Reviewer: ```suggestion
By default, Tune reports experiment progress to the command-line upon each training result received as follows.
```
